maestrowf/datastructures/dag.py

```diff
@@ -127,6 +127,35 @@ def remove_edge(self, src, dest):
         logging.debug("Removing edge (%s, %s).", src, dest)
         self.adjacency_table[src].remove(dest)
 
+    def __iter__(self):
+        """
+        Iterator over the graph
```

Comment:
This docstring could be clearer. It also looks like this is a DFS search. It might be worth mentioning that in the docstring or making it configurable somehow.